apps/web/src/features/safe-shield/components/TenderlySimulation.tsx

```diff
@@ -0,0 +1,330 @@
+    nestedTxInfo?.type === 'approveHash' && nestedTxInfo.signedHash && chain
+      ? {
+          chainId: chain.chainId,
+          txId: nestedTxInfo.signedHash,
+        }
+      : skipToken,
+  )
+
+  const [nestedSafeInfo] = useAsync(
+    () =>
+      !!chain && !!nestedTxInfo?.nestedSafeAddress
+        ? getSafeInfo(chain.chainId, nestedTxInfo.nestedSafeAddress)
+        : undefined,
+    [chain, nestedTxInfo],
+  )
+
+  const nestedSafeTx = useMemo<SafeTransaction | undefined>(() => {
+    if (!nestedTxInfo) return undefined
+
+    if (nestedTxInfo.type === 'approveHash' && nestedTxDetails) {
+      return {
+        addSignature: () => {},
+        encodedSignatures: () => '',
+        getSignature: () => undefined,
+        data: extractTxInfo(nestedTxDetails).txParams,
+        signatures: new Map(),
+      }
+    }
+
+    return undefined
+  }, [nestedTxInfo, nestedTxDetails])
+
+  const handleToggleSimulation = () => {
+    setSimulationExpanded(!simulationExpanded)
+  }
+
+  const handleRunSimulation = () => {
+    if (!safeTx) return
+
+    const executionOwner = isSafeOwner && signer?.address ? signer.address : safe.owners[0].value
+
+    const simulationParams = {
+      safe,
+      executionOwner,
+      transactions: safeTx,
+      gasLimit: undefined,
+    } as SimulationTxParams
+
+    simulation.simulateTransaction(simulationParams)
+
+    if (nestedTxInfo && nestedSafeInfo && nestedSafeTx) {
+      const nestedSimulationParams = {
+        safe: nestedSafeInfo,
+        executionOwner: safeAddress,
+        transactions: nestedSafeTx,
+        gasLimit: undefined,
+      } as SimulationTxParams
+
+      nestedTx.simulation.simulateTransaction(nestedSimulationParams)
+    }
+
+    setSimulationExpanded(true)
+  }
+
+  const isNested = !!nestedTxInfo && !!nestedSafeInfo && !!nestedSafeTx
+
+  const mainIsFinished = status.isFinished
+  const nestedIsFinished = isNested ? nestedTx.status.isFinished : true
+  const isSimulationFinished = mainIsFinished && nestedIsFinished
+
+  const mainIsSuccess = status.isSuccess && !status.isError
+  const nestedIsSuccess = isNested ? nestedTx.status.isSuccess && !nestedTx.status.isError : true
+  const isSimulationSuccess = mainIsSuccess && nestedIsSuccess
+
+  const isLoading = status.isLoading || (isNested && nestedTx.status.isLoading)
+
+  if (!showSimulation) {
+    return null
+  }
+
+  return (
+    <Box>
+      <Stack
+        direction="row"
+        justifyContent="space-between"
+        alignItems="center"
+        sx={{ padding: '12px', cursor: isSimulationFinished ? 'pointer' : 'default' }}
+        onClick={isSimulationFinished ? handleToggleSimulation : undefined}
+      >
+        <Stack direction="row" alignItems="center" gap={1}>
+          {isSimulationFinished ? (
+            isSimulationSuccess ? (
+              <CheckIcon sx={{ fontSize: 16, color: SEVERITY_COLORS.OK.main }} />
+            ) : (
+              <SvgIcon
```

Comment:
Can we reuse the SeverityIcon component here? And also for the replace the CheckIcon above.
https://github.com/safe-global/safe-wallet-monorepo/blob/tenderly-widget/apps/web/src/features/safe-shield/components/SeverityIcon.tsx/#L10
Then you could simply do this:

```suggestion
              <SeverityIcon severity={isSimulationSuccess ? Severity.OK : Severity.CRITICAL} />
```